src/stores/grid_renderer_store.ts

```diff
@@ -33,72 +35,41 @@ import {
   overlap,
   positionToZone,
   union,
-} from "../../helpers/index";
+} from "../helpers/index";
+import { Get } from "../store_engine";
 import {
   Align,
   Box,
   CellPosition,
   CellValueType,
   Dimension,
   GridRenderingContext,
-  HeaderDimensions,
   HeaderIndex,
   LAYERS,
   Pixel,
   UID,
   Viewport,
   Zone,
-} from "../../types/index";
-import { UIPlugin } from "../ui_plugin";
-
+} from "../types/index";
 // -----------------------------------------------------------------------------
 // Constants, types, helpers, ...
 // -----------------------------------------------------------------------------
 
 export const CELL_BACKGROUND_GRIDLINE_STROKE_STYLE = "#111";
 
-export class RendererPlugin extends UIPlugin {
-  static layers = [LAYERS.Background, LAYERS.Headers];
-  static getters = ["getColDimensionsInViewport", "getRowDimensionsInViewport"] as const;
-
-  private boxes: Box[] = [];
-
-  // ---------------------------------------------------------------------------
-  // Getters
-  // ---------------------------------------------------------------------------
-
-  /**
-   * Returns the size, start and end coordinates of a column relative to the left
-   * column of the current viewport
-   */
-  getColDimensionsInViewport(sheetId: UID, col: HeaderIndex): HeaderDimensions {
-    const left = Math.min(...this.getters.getSheetViewVisibleCols());
-    const start = this.getters.getColRowOffsetInViewport("COL", left, col);
-    const size = this.getters.getColSize(sheetId, col);
-    const isColHidden = this.getters.isColHidden(sheetId, col);
-    return {
-      start,
-      size: size,
-      end: start + (isColHidden ? 0 : size),
-    };
+export class GridRendererStore extends SpreadsheetStore {
+  constructor(get: Get) {
+    super(get);
+    markRaw(this);
```

Comment:
not sure about this (if it should be here or jus tin the renderedstore) - if both renderedStore and this one are reactive, then the time spent in reactive becomes huge, like 20+ times slower...
Should we have different type of spreadsheet stores? reactive and not reactive?